tudat-team/tudat
# Use Conda-Forge's CSPICE Package in Tudat

Please review and merge, @DominicDirkx. With this update, we will no longer need to maintain our own `cspice` fork and can instead utilize the peer-reviewed package maintained by astrojuanlu and AndrewAnnex available at [conda-forge/cspice-feedstock](https://github.com/conda-forge/cspice-feedstock). This change is long overdue. Re-rendering our own `cspice` was causing issues, and switching to the `conda-forge` package has proven to be a much faster and more sustainable solution.

### Testing Procedure
I followed these steps to ensure that the integration of the conda-forge `cspice` package works correctly with Tudat:

1. Created a new conda environment using the `environment.yaml` provided in the Tudat repository.
   ```bash
   conda env create -f environment.yml
   conda activate tudat
   ```

2. Set up the build directory and configured the project using CMake:
   ```bash
   mkdir build
   cd build
   cmake .. -DCMAKE_PREFIX_PATH=${CONDA_PREFIX}
   ```

3. Built the specific test target and ran it using `ctest`:
   ```bash
   cmake --build . --target test_spice_SpiceInterface
   ctest -R test_spice_SpiceInterface
   ```

   **Test Results:**
   ```
   Test project /home/geoffrey/tudat/build
       Start 205: test_spice_SpiceInterface
   1/1 Test #205: test_spice_SpiceInterface ........   Passed    0.03 sec

   100% tests passed, 0 tests failed out of 1

   Total Test time (real) =   0.05 sec
   ```

These results confirm that the `cspice` library from conda-forge is compatible with our current Tudat setup and meets all our requirements. This switch should simplify our dependencies and reduce maintenance overhead.
